### DISCUSSION TITLE
想问一下，怎么实现连线的时候，所有连接桩都展示出来啊

### DISCUSSION THREAD

**Discussion by Jesse0810** [antvis/X6#4115] [Q&A]
我看文档里面没有给连接过程中的监听事件，我都没法手动展示

**Comment by XueMeijing** [comment]
我感觉这里有点问题，连线肯定需要连接桩，所以是先展示连接桩才能连线。官方demo https://x6.antv.antgroup.com/examples/showcase/practices/#flowchart 中是鼠标移入节点展示连接桩。
```javascript
// 控制连接桩显示/隐藏
const showPorts = (ports: NodeListOf<SVGElement>, show: boolean) => {
  for (let i = 0, len = ports.length; i < len; i += 1) {
    ports[i].style.visibility = show ? 'visible' : 'hidden'
  }
}
graph.on('node:mouseenter', () => {
  const container = document.getElementById('graph-container')!
  const ports = container.querySelectorAll(
    '.x6-port-body',
  ) as NodeListOf<SVGElement>
  showPorts(ports, true)
})
graph.on('node:mouseleave', () => {
  const container = document.getElementById('graph-container')!
  const ports = container.querySelectorAll(
    '.x6-port-body',
  ) as NodeListOf<SVGElement>
  showPorts(ports, false)
})
```

**Comment by mrlionline** [comment]
同问，没有连线中事件监听，想实现从连接桩拖出线时可连接的桩高亮，不能连接的桩半透明。目前没有找到解决方案

**Reply by Yeti-xxx** [reply]
尝试 edge:added 这个事件？

**Reply by colonot** [reply]
graph.on("edge:added", ({ edge, index, options }) => {
        console.log("开始连接边");
      });
      graph.on("edge:connected", ({ isNew, edge }) => {
        console.log("连接边完成");
      });

**Comment by zlecheng** [comment]
你在从连接桩中拉出线的时候不就已经把画布上所有节点的连接桩显示出来了吗？

![image](https://github.com/user-attachments/assets/83f04954-dd62-4665-ae7e-a66cb4917912)
